solc.ts

```diff
@@ -78,7 +80,7 @@ function readFileCallback(sourcePath) {
 }
 
 function withUnixPathSeparators(filePath) {
-  if (os.platform !== 'win32')
+  if (os.platform as any !== 'win32')
```

Comment:
Why is this any?

Reply:
os.platform is actually a callable method and has been resolved to be called, fixing the need for the type conversion here.

Reply:
So this was a bug we had all along, and it just worked on unix because `function != win32`.

I know this sounds like a chore, but for traceability and binary search of changes, it is useful to separate things out a bit. This current PR is called "Typescript support" yet it fixes many other things.

Could you create a PR which just fixes this platform issue? Will merge it quickly.

Reply:
Yeah I will do this 12-1 GMT today.